### PR TITLE
Run `upstream` build with Python 3.10

### PR DESCRIPTION
**Pull request by jrbourbeau**
Possibly closes https://github.com/dask/dask/issues/9654

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
Alright, the changes here get out `upstream` CI build running again. Assuming there are no related failures, I'll plan to merge this PR after CI finishes as I suspect the changes here aren't particularly controversial and it'd be good to get our `upstream` build running ASAP. Happy to address any comments folks have in a follow-up PR

File: continuous_integration/environment-3.10.yaml [hunk 1]
```diff
@@ -68,5 +68,8 @@ dependencies:
   - mmh3
   - jinja2
   - pip
+  # The nightly pyarrow / arrow-cpp packages currently don't install with latest
```
Comment (ncclementi) [inline continuous_integration/environment-3.10.yaml:71]: Is this still a problem, I assume you checked but asking just in case.

**Review by ncclementi — APPROVED** [review]
This LGTM

File: continuous_integration/environment-3.10.yaml (hunk 1 above)
Comment (jrbourbeau) [reply continuous_integration/environment-3.10.yaml:71]: Thanks for reviewing @ncclementi. Yeah, this is still an issue. I initially forgot to transfer this change over to the 3.10 environment file and saw the solve issue from https://github.com/dask/dask/pull/9465 show up

**Pull request merged**